calculator/calculator.py

```diff
@@ -1,218 +1,112 @@
 """Calculator module.
 
 This module allows evaluation of strings which represent mathematical
-expressions. The supported operations are declared in operators dictionary.
-
-Example:
-    To use this module, import the evaluate() function:
-
-        from calculator import evaluate
-
-    The parameter of this function is a mathematical expression (str). For
-    example, "1 + (3^-2) * (4! - 5$ 2)".
+expressions. The supported operators are declared in operators dictionary.
 """
-
-import re
 import math
 import operator
+import re
+from collections import namedtuple
 from sys import maxsize
 
-any_number_regex = r"-?\d+(\.\d+)?"
-
-
-class Operator:
-    """Representation of a single mathematical operator.
-
-    Attributes:
-        operation (function): the function that does the operation.
-        precedence (int): the priority of the operation.
-        regex (str): regex to find a pattern which suits to the operator.
-
-    Note:
-        Precedence determines which operation goes after another.
-    """
-
-    def __init__(self, operation, precedence, regex):
-        self.operation = operation
-        self.precedence = precedence
-        self.regex = re.compile(regex)
-
-
-def average(num1, num2):
-    """Return the average of two numbers (int/float).
-
-    Note:
-        Same as using mean() from statistics module.
-    """
-    return float(num1 + num2) / 2.0
-
-
-def remove_spaces(expression: str) -> str:
-    """Remove the spaces between characters in a string."""
-    return ''.join(expression.split())
-
-
-def get_sub_expression_by_symbol(symbol: str, expression: str) -> str:
-    """Find the sub expression the current operator.
-
-    Args:
-        symbol: the current operator symbol.
-        expression: the entire expression to search.
-
-    Returns:
-         The desired sub expression of the operator. If nothing is found
-         returns an empty string.
-    """
-    match = operators[symbol].regex.search(expression)
-
-    if not match:
-        return ""
-
-    return match.group()
-
-
-def check_operands(func):
-    """Check the existence of the operands.
-
-    Args:
-        func (function): the function to operate on.
-
-    Returns:
-        function: the function wrapper.
-    """
-
-    def operands_wrapper(operator_symbol, operands):
-        left_operand, right_operand = operands
-
-        if left_operand and right_operand:
-            return func(operator_symbol, operands)
-
-        elif not left_operand and right_operand:
-            # if any([operator_symbol == "+", operator_symbol == "-"]):
-            #     return float(f"{operator_symbol}{right_operand}")
-            return func(operator_symbol, [right_operand])
-
-        elif not right_operand and left_operand:
-            return func(operator_symbol, [left_operand])
-
-    return operands_wrapper
-
+ANY_NUMBER = r"-?\d+(?:\.\d+)?"
+USER_MSG = "Enter expression to evaluate (enter 'quit' to exit) >>> "
```

Comment:
not a real global but a local variable, and shouldnt be a variable anyway I think

Reply:
Agree about the local variable but I think it should be a variable.